apps/whale-api/src/module.indexer/model/dftx/pool.swap.aggregated.ts

```diff
@@ -79,6 +80,25 @@ export class PoolSwapAggregatedIndexer extends DfTxIndexer<PoolSwap> {
 
   async invalidateTransaction (_: RawBlock, __: DfTxTransaction<PoolSwap>): Promise<void> {
   }
+
+  async getPoolPairs (block: RawBlock): Promise<PoolPairInfoWithId[]> {
+    const poolPairs = await this.rpcClient.poolpair.listPoolPairs({
+      start: 0,
+      limit: 1000000,
+      including_start: true
+    })
+    return Object.entries(poolPairs)
+      .map(([id, pair]): PoolPairInfoWithId => {
+        return { id, ...pair }
+      })
+      .filter(pair => {
+        return pair.creationHeight.lte(block.height)
+      })
+      .sort((a, b) => {
+        // Need to be sort by descending order otherwise aggregate indexer bucket will exit without creating a bucket
+        return b.creationHeight.comparedTo(a.creationHeight)
+      })
```

Comment:
```suggestion
    return Object.entries(poolPairs)
      .filter(pair => {
        return pair.creationHeight.lte(block.height)
      })
      .map(([id, pair]): PoolPairInfoWithId => {
        return { id, ...pair }
      })
      .sort((a, b) => {
        // Need to be sort by descending order otherwise aggregate indexer bucket will exit without creating a bucket
        return b.creationHeight.comparedTo(a.creationHeight)
      })
```

I would filter before doing the map to avoid copying unnecessary objects